### PR TITLE
upgrade some critical dependencies

### PR DESCRIPTION
**Pull request by krancour**
Depends on https://github.com/brigadecore/brigade/pull/1075 being merged first so that `go.mod` in this PR can be made to reference a formal release of brigade by semver instead of referencing a specific commit from my fork by sha.

### CODE REVIEW

**Comment by krancour** [comment]
glog used to use that flag, but I noticed the newer k8s libraries use klog, which is a fork of glog. I'll take a closer look later. Also, I am starting to wonder if ee shouldn't just upgrade all the way to the latest k8s libraries. Here, I just gave the same lib bumps that brigade proper got a while back, but there are newer libs since then. I'll look at bumping that all-around.

File: Makefile [hunk 1]
```diff
@@ -67,7 +67,7 @@ redeploy:
 
 .PHONY: dep
 dep:
-	$(DOCKER_CMD) dep ensure -v
+	$(DOCKER_CMD) go mod tidy && go mod vendor
```
Comment (vdice) [inline Makefile:70]:
```suggestion
	$(DOCKER_CMD) sh -c "go mod tidy && go mod vendor"
```

Comment (krancour) [reply Makefile]: Fixed in the latest commit

**Comment by krancour** [comment]
Just to circle back... the go.mod file here still needs a little update after brigade proper 1.4 is released so that we can reference a formal release instead of a git sha.

**Pull request merged**